### PR TITLE
Remove VBeam Speed Hack.

### PR DESCRIPTION
**Pull request by phire**
As far as I can see, VBeam Speed Hack was an attempt to reduce overhead only rendering one of odd/even fields every frame (since for a 30fps game, they are identical anyway).

But most of this code is missing and the rest is unmaintained, so the only effect of enabling vbeam speed hack is to mess up a hole bunch of audio and video timings. The intended side effects are to run the audio interrupt twice as often, and run the vbeam update function half as often, but there appear to be plenty of unintended side effects too.

Generally, one of two things happens to people who enable vbeam speed hack:
 1. They end up on the forums, IRC or our bug tracker complaining about audio bugs.
 2. They try and develop 60fps hacks.

VBeam Speed Hack is not the correct direction for 60fps hacks. It is not an old unused feature, it is an ugly bug.
Instead I recommend the development of better AR codes. Instead of trying to slow the game down after you somehow manage to get it rendering at double speed, you need to modify the VI interrupt so it fires twice a frame instead of once a frame (once for each of the odd and even fields) and the game renders twice as many frames. Overclock the CPU as necessary for the extra workload.

Such an AR code would theoretically work on actual hardware, if the GPU and/or CPU were fast enough.

### CODE REVIEW

File: Source/Core/Core/HW/VideoInterface.cpp [hunk 1]
```diff
@@ -478,10 +478,7 @@ void UpdateParameters()
 
 int GetNumFields()
 {
-	if (SConfig::GetInstance().m_LocalCoreStartupParameter.bVBeamSpeedHack)
-		return (2 / fields);
-	else
-		return 1;
+	return 1;
```
Comment (degasus) [inline Source/Core/Core/HW/VideoInterface.cpp]: There is no need for a function which just always return "1".

Comment (phire) [inline Source/Core/Core/HW/VideoInterface.cpp]: Yeah, I'm not actually sure what this function is for, NumFields What?

Comment (degasus) [inline Source/Core/Core/HW/VideoInterface.cpp]: Likely part of this hack to also slow down other functions, eg audio output ...

Comment (phire) [inline Source/Core/Core/HW/VideoInterface.cpp]: Yeah, that's likely. 
Removed.

File: Source/Core/Core/HW/SystemTimers.cpp [hunk 2]
```diff
@@ -242,7 +241,7 @@ void Init()
 
 		// FYI, WII_IPC_HLE_Interface::Update is also called in WII_IPCInterface::Write32
 		const int freq = 1500;
-		IPC_HLE_PERIOD = GetTicksPerSecond() / (freq * VideoInterface::GetNumFields());
+		IPC_HLE_PERIOD = GetTicksPerSecond() / (freq);
```
Comment (degasus) [inline Source/Core/Core/HW/SystemTimers.cpp]: braces

**Comment by degasus** [comment]
LGTM

**Comment by neobrain** [comment]
<3

**Comment by phire** [comment]
@dolphin-emu-bot rebuild

**Comment by delroth** [comment]
\o/

Also, dat branch name <3

**Pull request merged**